src/Client/HedgedConnections.cpp

```diff
@@ -177,6 +177,10 @@ void HedgedConnections::sendQuery(
     {
         Settings modified_settings = settings;
 
+        // Kusto queries to replicas are transformed to ClickHouse-SQL. Change the setting before sending.
+        if (modified_settings.dialect == Dialect::kusto)
```

Comment:
This looks wrong. Why only Kusto?
What about the PRQL dialect?

Reply:
I will investigate. I hadn't seen any issues submitted for PRQL having the same issue with distributed tables, and thought maybe the dialect was doing something else making it unaffected. I will report what I find.

Reply:
I can confirm that this issue also effects PRQL. Ex. error output

```
Received exception from server (version 23.12.2):
Code: 62. DB::Exception: Received from localhost:9000. DB::Exception: Received from clickhouse03:9000. DB::Exception: PRQL syntax error: 'Error:
   ╭─[:1:1]
   │
 1 │ SELECT `shared_test_table`.`id` FROM `default`.`shared_test_table`
   │ ───┬──
   │    ╰──── Unknown name
───╯
'. (SYNTAX_ERROR)
```

Here, we can see the same behavior, of the transformed SQL statement trying to be executed in the incorrect dialect.

I will extend my change and add appropriate tests.